manifests/init.pp

```diff
@@ -1,9 +1,11 @@
 # @summary Install and configure zram_generator zram mounts
 # @see https://github.com/systemd/zram-generator
 # @param install_defaults Controls if the package 'zram-generator-defaults' be installed.
+# @param manage_defaults_package wether the zram-generator-defaults package should be installed
 #
 class zram_generator (
   Enum['installed', 'absent'] $install_defaults = 'absent',
+  Boolean $manage_defaults_package = $facts['os'] ? { 'Archlinux' => false, default => true },
```

Comment:
You can even consider not exposing it and only keep it as an internal private variable (thus only leaving it to Hiera), but at least correct the logic:
```suggestion
  Boolean $manage_defaults_package = $facts['os']['name'] != 'Archlinux',
```